File: Backend/Ecommerce/Models/Dtos/SearchDto.cs
namespace Ecommerce.Models.Dtos;

// aqui es la busqueda que hago desde el cliente y se la paso al servicio, que hará la consulta 

public class SearchDto
{
    public string consulta { get; set; } // lo que buscas
                                         
    public enum CriterioOrden  
    {
        Name,
        Price
    }
    public CriterioOrden Criterio { get; set; }
    public bool Orden { get; set; } // o tamb enum, asc o desc
    public int PaginaActual  { get; set; }

}

File: Frontend/src/app/models/searchDto.ts
export interface SearchDto{
    consulta: string; 
    Criterio: CriterioOrden; 
    Orden: boolean;   // true asc, false desc
    PaginaActual: number; 
  }

  export enum CriterioOrden {
    Name = 0, 
    Price = 1
}